src/components/SupportRequestModal.js

```diff
@@ -14,91 +13,96 @@ import validate from 'validate.js'
-        onChange: e => this.setState({ description: e.target.value })
-      }),
-      Forms.requiredFormLabel('Contact email'),
-      validatedInput({
-        inputProps: {
+      return h(Modal, {
+        onDismiss,
+        title: 'Contact Us',
+        okButton: buttonPrimary({
+          disabled: errors,
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: () => this.submit()
+        }, ['SEND'])
+      }, [
+        !isSignedIn && Forms.requiredFormLabel('Name'),
+        !isSignedIn && textInput({
+          placeholder: 'What should we call you?',
+          autoFocus: true,
+          value: !isSignedIn ? nameEntered : firstName,
+          onChange: e => this.setState({ nameEntered: e.target.value })
+        }),
+        Forms.requiredFormLabel('Type'),
+        h(Select, {
+          isMulti: false,
+          value: type,
+          onChange: ({ value }) => this.setState({ type: value }),
+          options: [{ value: 'question', label: 'Question' }, { value: 'bug', label: 'Bug' }, { value: 'feature_request', label: 'Feature Request' }]
+        }),
+        Forms.requiredFormLabel(`How can we help you${greetUser}?`),
+        textInput({
+          style: { borderBottomLeftRadius: 0, borderBottomRightRadius: 0, borderBottomStyle: 'dashed' },
+          placeholder: 'Enter a subject',
+          autoFocus: !!isSignedIn,
+          value: subject,
+          onChange: e => this.setState({ subject: e.target.value })
+        }),
+        h(TextArea, {
+          style: { height: 200, borderTopLeftRadius: 0, borderTopRightRadius: 0, borderTopStyle: 'dashed' },
+          placeholder: 'Enter a description',
+          value: description,
+          onChange: e => this.setState({ description: e.target.value })
+        }),
+        Forms.requiredFormLabel('Contact email'),
+        textInput({
           value: email,
+          placeholder: 'Enter your email address',
           onChange: e => this.setState({ email: e.target.value })
-        },
-        error: Utils.summarizeErrors(errors && errors.email)
-      }),
-      submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.red[0] } }, [submitError]),
-      submitting && spinnerOverlay
-    ])
-  }
+        }),
+        submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.red[0] } }, [submitError]),
+        submitting && spinnerOverlay
+      ])
+    }
 
-  async submit() {
-    const { onSuccess, ajax: { User }, authState: { profile: { firstName, lastName } } } = this.props
-    const { email, type, description, subject } = this.state
-    const name = `${firstName} ${lastName}`
-    const currUrl = window.location.href
+    async submit() {
+      const { onSuccess, authState: { isSignedIn, profile: { firstName, lastName } } } = this.props
+      const { email, type, description, subject, nameEntered } = this.state
+      const name = isSignedIn ? `${firstName} ${lastName}` : nameEntered
```

Comment:
This code is using first + last, while the validation above is just using first name. Since we want to make sure that it's consistent, and the logic has become non-trivial, it might be good to extract a helper method that assembles the full object from props/state, which can be used in both places.